### ISSUE TITLE
Jumper modeling

### ISSUE THREAD

**Issue by jinningwang** [CURENT/ams#96]
**Describe the bug**

Support the model `Jumper` (called `Switch` sometimes)
 
**To Reproduce**

Update data files and code for reproducing the error.

If you need to insert code inline, use a pair of triple backticks to format:

    ```python
    Code goes here
    ```

**Expected behavior**

- Support Jumper model thus two `Bus` can connect together
- It would be good if the system matrices (PTDF, etc.) remain the same shape, but not sure if easy to do
- For valid Jumper (conditions check passed), actions to take when altering `u`: 1) update system matrices 2) anything else?

**Desktop (please complete the following information):**

 - OS: [e.g. Windows]
 - AMS version (please paste the output from `ams misc --version`)

**pip packages (please paste the output from `pip list`)**

**Additional context**

Proposed development efforts:
1. Add model `Jumper`
1. Add system connectivity check
1. Add action flow to allow automatic switching action (open -> close, close -> open)
1. Improve ANDES interface to support this
1. Improve MATPOWER file converter to support this

Reference:
[ANDES - Jumper](https://docs.andes.app/en/latest/groupdoc/ACShort.html#jumper)
[pandapower - switch](https://pandapower.readthedocs.io/en/latest/elements/switch.html#electric-model)

**Comment by jinningwang** [comment]
@cuihantao Hantao, I summarized the Jumper development issues here and welcome your inputs.

Before actual development efforts, I need to figure out a doable plan.

**Comment by cuihantao** [comment]
Some of the pieces are already there in ANDES. There is a Jumper model and a connectivity check algorithm. Might need to be specific about the gaps.

**Comment by jinningwang** [comment]
Some suggestions received today: 1) lazy evaluation pattern, a status flag to indicate the topology change, and do the actual action when they are used, 2) keeping records can be a headache

**Comment by cuihantao** [comment]
Yeah, if you would like my comments, I will need to know some specifics.

**Comment by leocuisine** [comment]
Closing — the work in this issue has shipped. Summary of what's live as of v1.2.3 (2026-05-02):

**Model.** `Jumper` is implemented in [`ams/models/line.py`](https://github.com/CURENT/ams/blob/master/ams/models/line.py#L53), inheriting `JumperData` + `Model`, with the full short-two-buses semantics documented in the class docstring (voltage-control compatibility rules, `u`-toggle behavior, `(p, q)` flow direction).

**System connectivity.** `System.connectivity()` covers the split-detection action flow called out in the docstring (`Setting a Jumper's connectivity status u to zero will disconnect the two buses... call System.connectivity() immediately following the split to detect islands`).

**ANDES interface.** `to_andes()` carries the `Jumper` device through the AMS→ANDES bridge — covered by [`tests/interop/test_andes_interface.py:276`](https://github.com/CURENT/ams/blob/master/tests/interop/test_andes_interface.py#L276) (`TestJumperToAndes`, migrated from the original `tests/test_jumper.py` during the v1.2.3 tests reorganization).

**MATPOWER converter + case data.** `pjm5bus_jumper.xlsx` ships under `ams/cases/` (see [`ams/cases/README.md`](https://github.com/CURENT/ams/blob/master/ams/cases/README.md)) and the to_mpc / from_mpc paths handle it.

The remaining open piece from this issue — automatic open↔close action flow — is better tracked as its own focused issue if/when needed; the current `u`-toggle plus `connectivity()` covers the actionable user pattern.

Mapping to the original task list:
- [x] Add model `Jumper`
- [x] Add system connectivity check (`System.connectivity()`)
- [ ] Automatic open→close / close→open switching action — deferred; open separately if needed
- [x] ANDES interface support (`to_andes()` + `TestJumperToAndes`)
- [x] MATPOWER converter support (jumper case shipped, `to_mpc` / `from_mpc` paths exercised)
